app/javascript/components/automate-simulation-form/index.jsx

```diff
@@ -0,0 +1,114 @@
+}) => {
+  const typeClassesOptions = [
+    { label: `<${__('None')}>`, value: '-1' },
+    ...Object.entries(resolve.target_classes).map(([key, value]) => ({ label: value, value: key })),
+  ];
+  const headers = {
+    'Content-Type': 'application/json',
+  };
+
+  const [formData, setFormData] = useState({
+    isLoading: false,
+    tempData: undefined,
+    targetClass: '-1',
+    simulationTree: { notice: 'Enter Automation Simulation options on the left and press Submit' },
+  });
+
+  useEffect(() => {
+    if (formData.isLoading) {
+      http.post(url, JSON.stringify(formData.tempData), { headers })
+        .then((result) => {
+          setFormData({
+            ...formData,
+            isLoading: false,
+            simulationTree: result,
+          });
+        })
+        .catch((error) => console.log('error: ', error));
+    }
+  }, [formData.isLoading]);
+
+  const handleSubmit = (values) => {
+    const instanceName = Array.isArray(values.instance_name) ? values.instance_name[0] : values.instance_name;
+    const data = {
+      instance_name: instanceName,
+      object_message: values.object_message,
+      object_request: values.object_request,
+      target_class: values.target_class,
+      readonly: values.readonly,
+      selection_target: values.selection_target,
+      button: 'throw',
+    };
+
+    const attributes = Array.from({ length: attrValuesPairs.length }, (_, i) => i + 1).flatMap((i) => [`attribute_${i}`, `value_${i}`]);
+    const attrValPairs = Object.fromEntries(
+      attributes.flatMap((key) => (values[key] ? [[key, values[key]]] : []))
+    );
+
+    Object.entries(attrValPairs).forEach(([key, value]) => {
+      data[key] = value;
+    });
+
+    setFormData({
+      ...formData,
+      isLoading: true,
+      tempData: data,
+    });
+  };
+
+  const onFormReset = () => {
+    const buttons = document.querySelectorAll('.bx--list-box__selection');
+
+    for (let i = 0; i < buttons.length; i += 1) {
+      const resetButton = buttons[i];
+      resetButton.click();
+    }
+    document.getElementById('object_request').value = '';
+  };
+
+  return (
+    <div className="automate-simulation-page">
+      <div className="automate-simulation-form-wrapper">
+        <MiqFormRenderer
+          className="automate-simulation-form"
+          schema={createSchema(
+            resolve, maxNameLength, url, aeAnsibleCustomButton, formAction, aeCustomButton, attrValuesPairs,
+            maxLength, typeClassesOptions, formData, setFormData
+          )}
+          onSubmit={handleSubmit}
+          canReset
+          onReset={onFormReset}
+        />
+      </div>
+      <div className="automate-simulation-summary-wrapper">
+        <div className="simulation-title-text">
+          Simulation
+        </div>
+        <div className="automate-simulation-summary">
+          <AutomationSimulation data={formData.simulationTree !== undefined ? formData.simulationTree : {}} />
```

Comment:
also, could you see if we can make use of the isLoading variable to display a loader component, since we are using the api response to display the results.